api_v3/services/LiveStreamService.php

```diff
@@ -575,5 +574,97 @@ public function regenerateStreamTokenAction($entryId)
 		$entry = KalturaEntryFactory::getInstanceByType($liveEntry->getType());
 		$entry->fromObject($liveEntry, $this->getResponseProfile());
 		return $entry;
+
 	}
+
+    /**
+     *
+     * @action archive
+     * @param string $liveEntryId
+     * @return bool
+     * @throws KalturaAPIException
+     * @throws KalturaClientException
+     * @throws PropelException
```

Comment:
catch the propelException and if we must thrown an KalturaAPIException, with a relevant message, instead